homeassistant/components/flux_led/light.py

```diff
@@ -171,133 +188,222 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
+        device: dict,
+        effect_speed: int,
+    ):
+        """Initialize the Flux light entity."""
+        super().__init__(coordinator=coordinator)
+
+        self._name = device[CONF_NAME]
+        self._unique_id = unique_id
+        self._icon = "mdi:lightbulb"
+        self._attrs = {}
+        self._state = None
+        self._brightness = None
+        self._hs_color = None
+        self._bulb = coordinator.light
+        self._last_brightness = 255
+        self._last_hs_color = color_util.color_RGB_to_hs(255, 255, 255)
+        self._ip_address = device[CONF_HOST]
+        self._effect_speed = effect_speed
+        self._mode = coordinator.data["mode"]
+
+    async def async_remove_light(self, device: dict):
+        """Remove a bulb device when it is removed from options."""
+
+        bulb_id = device["device_id"]
+
+        if self._unique_id != bulb_id:
+            return
 
-    def _disconnect(self):
-        """Disconnect from Flux light."""
-        self._bulb = None
+        entity_registry = await self.hass.helpers.entity_registry.async_get_registry()
+        entity_entry = entity_registry.async_get(self.entity_id)
+
+        device_registry = await self.hass.helpers.device_registry.async_get_registry()
+        device_entry = device_registry.async_get(entity_entry.device_id)
+
+        if (
+            len(
+                async_entries_for_device(
+                    entity_registry,
+                    entity_entry.device_id,
+                    include_disabled_entities=True,
+                )
+            )
+            == 1
+        ):
+            # If only this entity exists on this device, remove the device.
+            device_registry.async_remove_device(device_entry.id)
+
+        entity_registry.async_remove(self.entity_id)
+
+    async def async_added_to_hass(self):
+        """Run when the entity is about to be added to hass."""
+        await super().async_added_to_hass()
+
+        async_dispatcher_connect(
+            self.hass, SIGNAL_REMOVE_DEVICE, self.async_remove_light
+        )
 
     @property
-    def available(self) -> bool:
-        """Return True if entity is available."""
-        return self._bulb is not None
+    def unique_id(self):
+        """Return the unique ID of the light."""
+        return self._unique_id
 
     @property
     def name(self):
-        """Return the name of the device if any."""
+        """Return the name of the light."""
         return self._name
 
     @property
     def is_on(self):
-        """Return true if device is on."""
-        return self._bulb.isOn()
+        """Return true if the light is on."""
+
+        return self._state
 
     @property
     def brightness(self):
-        """Return the brightness of this light between 0..255."""
-        if self._mode == MODE_WHITE:
-            return self.white_value
+        """Return the brightness of the light."""
+
+        self._brightness = self.coordinator.data["brightness"]
```

Comment:
If we update state holding instance attributes in the state properties we'll overwrite the changes we saved in the actuator methods when the state updates. We don't want that.

Only update the state holding instance attributes from the coordinator via the callback that is attached to the coordinator.

Reply:
Not sure I understand. So I don't need to update `self._brightness` from the data in the coordinator in the properties? Where would that get set then? I don't see anywhere in the `CoordinatorEntity` where it is updating `self._brightness` from the coordinator.

Are you suggesting I just remove the line: 
`self._brightness = self.coordinator.data["brightness"]`

Reply:
Implement `_handle_coordinator_update` in the entity and update the attributes from the coordinator data.

Reply:
OK - think I got it. Make sense to do all the updates in one place. Much simpler.